platform/macos/app/MapDocument.m

```diff
@@ -196,16 +197,21 @@ - (IBAction)takeSnapshot:(id)sender {
                         
                     }
                     
-                    NSString *extension = [[fileURL pathExtension] lowercaseString];
-                    NSBitmapImageFileType fileType;
-                    if ([extension isEqualToString:@"png"]) {
-                        fileType = NSPNGFileType;
-                    } else if ([extension isEqualToString:@"gif"]) {
-                        fileType =  NSGIFFileType;
-                    } else if ([extension isEqualToString:@"jpg"] || [extension isEqualToString:@"jpeg"]) {
-                        fileType =  NSJPEGFileType;
-                    } else {
-                        fileType =  NSTIFFFileType;
+                    NSString *uti;
+                    [fileURL getResourceValue:&uti forKey:NSURLTypeIdentifierKey  error:nil];
+                    NSBitmapImageFileType fileType = NSTIFFFileType;
+                    
+                    for (NSString *imageType in [NSImage imageTypes]) {
+                        if ([uti isEqualToString:imageType]) {
+                            if ([uti containsString:@"png"]) {
```

Comment:
`public.png` isn’t the only UTI that contains the substring `png`, and UTIs that don’t contain `png` may still be PNGs. Use `UTTypeConformsTo()` from the Core Services framework to perform this check:

```objc
UTTypeConformsTo((__bridge CFStringRef)uti, kUTTypePNG)
```